extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-attachment.js

```diff
@@ -26,113 +28,205 @@ describes.realWin(
-          'productTitle': 'Brass Lamp',
-          'productBrand': 'Lamp Co',
-          'productPrice': 799.0,
-          'productPriceCurrency': 'USD',
-          'productImages': ['https://source.unsplash.com/Ry9WBo3qmoc/500x500'],
-        },
-      };
-      storeService.dispatch(Action.CHANGE_PAGE, {
-        id: 'page1',
-        index: 1,
+    function dispatchActiveProductData() {
+      storeService.dispatch(Action.ADD_SHOPPING_DATA, {
+        activeProductData: shoppingData.items[0],
       });
-      storeService.dispatch(Action.ADD_SHOPPING_DATA, shoppingData);
-      storeService.dispatch(Action.TOGGLE_PAGE_ATTACHMENT_STATE, true);
     }
 
     it('should build shopping attachment component', () => {
       expect(() => shoppingImpl.layoutCallback()).to.not.throw();
     });
 
-    it('should build CTA with i18n shopping label text', async () => {
-      await dispatchTestShoppingData();
-      const attachmentChildEl = shoppingEl.querySelector(
-        'amp-story-page-attachment'
-      );
+    it('should build CTA with i18n shopping label text', () => {
       expect(attachmentChildEl.getAttribute('cta-text')).to.equal('Shop Now');
     });
 
-    it('should open attachment', async () => {
-      await dispatchTestShoppingData();
-      const attachmentChildEl = shoppingEl.querySelector(
-        'amp-story-page-attachment'
-      );
-      const attachmentChildImpl = await attachmentChildEl.getImpl();
-      env.sandbox.stub(attachmentChildImpl, 'mutateElement').callsFake(() => {
-        expect(pageEl.querySelector('.i-amphtml-story-draggable-drawer-open'))
-          .to.not.be.null;
-      });
+    it('should build PLP on attachment state open if no active product data', async () => {
+      await shoppingImpl.layoutCallback();
+      await attachmentChildImpl.layoutCallback();
+      storeService.dispatch(Action.TOGGLE_PAGE_ATTACHMENT_STATE, true);
+      expect(
+        attachmentChildEl.querySelector('.i-amphtml-amp-story-shopping-plp')
+      ).to.not.be.null;
     });
 
-    it('should build PLP on CTA click', async () => {
-      await dispatchTestShoppingData();
-      const attachmentChildEl = shoppingEl.querySelector(
-        'amp-story-page-attachment'
-      );
-      const attachmentChildImpl = await attachmentChildEl.getImpl();
-      env.sandbox.stub(attachmentChildImpl, 'mutateElement').callsFake(() => {
-        expect(pageEl.querySelector('.amp-story-shopping-plp')).to.not.be.null;
-      });
+    it('should open attachment when active product data is set', async () => {
+      await shoppingImpl.layoutCallback();
+      await attachmentChildImpl.layoutCallback();
+      dispatchActiveProductData();
+      // Simulating the getImpl in amp-story-shopping-attachment's
+      // onShoppingDataUpdate_ method
+      await attachmentChildEl.getImpl();
+      expect(
+        attachmentChildEl.classList.contains(
+          'i-amphtml-story-draggable-drawer-open'
+        )
+      ).to.be.true;
```

Comment:
(I'm super nitty here) Testing  either`attachmentChildImpl.state === DrawerState.OPEN` or the store service ATTACHMENT_STATE might be a bit better. You'd expect being able to change the class name without breaking the tests